File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

vector<int> solution(int k, vector<int> score) {
    vector<int> answer;
    vector<int> v;
    for(int i = 0; i < score.size(); i++){
        v.push_back(score[i]);
        sort(v.begin(), v.end(), greater<int>{});
        if(v.size() >= k)
            answer.push_back(v[k-1]);
        else
            answer.push_back(v[v.size()-1]);
    }
    
    return answer;
}

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    stk = []
    for n in score:
        if len(stk) < k:
            stk.append(n)
            stk.sort(reverse=True)
            answer.append(stk[-1])
        else:
            stk.append(n)
            stk.sort(reverse=True)
            stk = stk[:k]
            answer.append(stk[-1])
    return answer